### DISCUSSION TITLE
Expected preloaded route behaviour

### DISCUSSION THREAD

**Discussion by roduyemi** [TanStack/router#1707] [Q&A] [answered]
Hello,
We're wondering what should happen when `preload` is set to `intent` on the router and we hover over a link?
We can see the pending / cached matches are updated but we were also expecting the component's code to load in at the same time.
May you please confirm?
Thanks!

**Comment by SeanCassiere** [comment, marked as answer]
@roduyemi this does happen, but you need to have components correctly code-split. The docs, have guides on code-splitting.

**Reply by roduyemi** [reply]
Thanks for confirming Sean! 🙏🏿
